File: lib/journal/Bland.cc
// -*- c++ -*-
//
// michael a.g. aïvázis <michael.aivazis@para-sim.com>
// (c) 1998-2023 all rights reserved


// external support
#include "externals.h"
// forward declarations
#include "forward.h"
// type aliases
#include "api.h"

// the global settings
#include "Chronicler.h"
// message contents
#include "Entry.h"
// the superclass
#include "Renderer.h"
// the declaration
#include "Bland.h"


// metamethods
pyre::journal::Bland::~Bland() {}


// implementation details
void
pyre::journal::Bland::header(
    palette_type & palette, linebuf_type & buffer, const entry_type & entry) const
{
    // nothing to do
    return;
}


void
pyre::journal::Bland::body(
    palette_type & palette, linebuf_type & buffer, const entry_type & entry) const
{
    // get the page
    auto & page = entry.page();

    // go through the lines in the page
    for (auto line : page) {
        // and render each one
        buffer << "  " << line << std::endl;
    }

    return;
}


void
pyre::journal::Bland::footer(
    palette_type & palette, linebuf_type & buffer, const entry_type & entry) const
{
    // nothing to do
    return;
}


// end of file

File: packages/journal/Bland.py
# -*- coding: utf-8 -*-
#
# michael a.g. aïvázis <michael.aivazis@para-sim.com>
# (c) 1998-2023 all rights reserved


# superclass
from .Renderer import Renderer


# the renderer or developer-facing messages
class Bland(Renderer):
    """
    The renderer of user-facing messages
    """


    # implementation details
    def body(self, palette, entry):
        """
        Generate the message body
        """
        # get the page
        page = entry.page
        # indent
        indent = " " * 2
        # go through the message content
        for line in page:
            # and print each line
            yield f"{indent}{line}"
        # all done
        return


# end of file
